fvm/transactionInvocator.go

```diff
@@ -152,54 +152,87 @@ func (i *TransactionInvocator) Process(
 	// 	panic(err)
 	// }
 
+	// try to deduct fees even if there is an error.
+	feesError := i.deductTransactionFees(env, proc)
+	if feesError != nil {
+		txError = feesError
+	}
+
 	// applying contract changes
 	// this writes back the contract contents to accounts
 	// if any error occurs we fail the tx
+	// this needs to happen before checking limits, so that contract changes are committed to the state
 	updatedKeys, err := env.Commit()
 	if err != nil && txError == nil {
 		txError = fmt.Errorf("transaction invocation failed: %w", err)
```

Comment:
So if we had `feesError` we ignore this newly assigned error here?

Reply:
That is correct. If there is a error with deducting fees (or with checking storage) the tx error should be hidden from the user. But we should probably still log it... I will add that.